### PR TITLE
[StructuralMechanicsApplication] Clean up cpp warnings

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR removes several warnings from cpp tests in StructuralMechanicsApplication

### CODE REVIEW

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_static_sensitivity.cpp [hunk 1]
```diff
@@ -97,8 +97,7 @@ KRATOS_TEST_CASE_IN_SUITE(TotalLagrangian2D3_StaticSensitivity, KratosStructural
     smtsss::PrimalTestSolver solver{&primal_model_part, response_node_id};
     const double delta = 1e-7;
     const double response_value0 = solver.CalculateResponseValue();
-    ModelPart& adjoint_model_part =
-        CreateStructuralMechanicsAdjointTestModelPart(&primal_model_part);
+    ModelPart& adjoint_model_part = CreateStructuralMechanicsAdjointTestModelPart(&primal_model_part);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_static_sensitivity.cpp:100]: what is the change here?

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp [hunk 2]
```diff
@@ -134,7 +134,7 @@ SolvingStrategyType::Pointer CreatePrimalSolvingStrategy(ModelPart* pModelPart)
     SchemeType::Pointer p_scheme =
         Kratos::make_shared<ResidualBasedBossakDisplacementScheme<SparseSpaceType, LocalSpaceType>>(0.0);
     ConvergenceCriteriaType::Pointer p_conv_criteria =
-        Kratos::make_shared<ResidualCriteria<SparseSpaceType, LocalSpaceType>>(1e-24, 1e-25);
+        Kratos::make_shared<ResidualCriteria<SparseSpaceType, LocalSpaceType>>(1e-12, 1e-14);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp:137]: why do you decrease the tolerance 

FYI @msandre

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp [hunk 3]
```diff
@@ -209,6 +209,8 @@ namespace Kratos
             process_info[STEP] = 1;
             process_info[NL_ITERATION_NUMBER] = 1;
             
+            Testing::Create2DGeometry(this_model_part, "SmallDisplacementElement2D3N");
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp:212]: what is the change here?

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp [hunk 4]
```diff
@@ -257,7 +257,9 @@ namespace Kratos
             auto& process_info = this_model_part.GetProcessInfo();
             process_info[STEP] = 1;
             process_info[NL_ITERATION_NUMBER] = 1;
-            
+
+            Testing::Create3DGeometry(this_model_part, "SmallDisplacementElement3D4N");
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp:261]: same

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_static_sensitivity.cpp (hunk 1 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_static_sensitivity.cpp:100]: Probably just one line jump, sorry for that

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp:137]: Run the current tests with the current tolerance, it doesn't converge. With this tolerance it converges and the test pass

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp (hunk 3 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp:212]: The create before the Get

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp (hunk 4 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_spr_error_process.cpp:261]: Idem

**Comment by philbucher** [comment]
ok for me, @msandre can you please have a look at your test?

**Comment by loumalouomega** [comment]
Ping @msandre

**Review by msandre — APPROVED** [review]
:+1: Thanks

**Pull request merged**